src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -77,23 +77,18 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
             // For now we do our best to handle as many cases as we can.
             // Tracking issue - https://github.com/dotnet/roslyn/issues/68083
 
-            // If the uri is a file then use the simplified absolute or local path.
-            // In other cases (e.g. git files) use the full uri string. This ensures documents
-            // with the same local path (e.g. git://someFilePath and file://someFilePath) are differentiated.
-            if (documentUri.IsFile)
-            {
-                var fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.AbsolutePath);
-                if (fileDocumentIds.Any())
-                    return fileDocumentIds;
+            // Use the original string the Uri was created with to avoid any Unicode escaping issues
+            // (see https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1842233).
+            // Only if the original string is not an existing document path and the Uri has "file" scheme,
+            // use the LocalPath, in case the client creates Uri instance with using the URL string "file://...".
 
-                fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
+            var fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.OriginalString);
```

Comment:
is original string correct?  When we get a vscode URI (e.g. file://c:/somePath) won't we be comparing the original string (file://c:/somePath) with the document file path (c:/somePath)?

Reply:
Yes, and it won't exist, so we fall back to `LocalPath`.

We should really just remove usage of `Uri` class from LSP.

Reply:
There is no issue with collisions since `file://...` paths are not valid file paths.

Reply:
And what if we get a non file path URI?  That is totally possible for vscode virtual documents.  For example I believe Razor files

Reply:
Then we use the OriginalString.

Reply:
Ah right I guess we store the full URI in the filepath in that case.  This seems probably safe then?

Reply:
whatever we are doing (or assuming) please just add some tests to validate that it works how we think it shoudl work.

Reply:
Whenever we use `FilePath` in Roslyn we should keep it as `string`. It can be either an absolute local path or a URL. These two formats never collide.

If the client is only using URL to represent document paths then all `Document.FilePath` properties will be in "file://..." format (or other scheme). If the client is using file paths then all documents have regular local file paths like today.
For virtual documents we can use URL format. 

The only problems might occur if we need to interpret the path. We should avoid doing that as much as possible. But if we need to extract file name or directory name then we can first determine whether the string is a URL. If so we use `Uri` class to parse out the file/directory name. Otherwise, we use regular `Path` API.